docs/docs/03-tools/05-context.md

```diff
@@ -87,11 +79,50 @@ context: ./context_with_arg.gpt with ${search} as search_tool
 Args: search: Search tool to use
 
 What are some of the most popular tourist destinations in Scotland, and how many people visit them each year?
+```
+
+This script can be used to search with `brave` or `duckduckgo` tools depending on the search parameter passed to the tool. Example usage for using the brave search tool:
 
+```sh
+gptscript --disable-cache my_context_with_arg.gpt '{"search": "brave"}'
 ```
 
-This script can be used to search with `brave` or `duckduckdb` tools depending on the search parameter passed to the tool.
-Example usage for using brave search tool:
+## Comprehensive Example of a Context Provider Tool
+
+Here is a more streamlined example of a context provider tool that aggregates multiple smaller contexts and tools:
+
 ```yaml
-gptscript --disable-cache my_context_with_arg.gpt '{"search": "brave"}'
+# tool.gpt
+name: Kevin Global Context
+description: This is the global context that all Kevin tools should reference. This context just aggregates a series of smaller contexts.
+
+share context: github.com/gptscript-ai/context/cli
+share context: ./cmds.gpt
+
+share tools: github.com/gptscript-ai/answers-from-the-internet
+
+#!sys.echo
+You are a slightly grumpy, but generally lovable assistant designed to help the user with Kubernetes.
+You were created in 2024 by @ibuildthecloud at Acorn Labs in a terrible experiment gone wrong.
+
+Guidelines:
+1. Always run tools sequentially and never in parallel.
+2. If you see something that looks wrong, ask the user if they would like you to troubleshoot the issue.
+3. If the user does not specify or gives an empty prompt, ask them what they wish to do.
+```
+If the above tool were in a directory called `context`, it could be referenced by a tool like this:
+```yaml
+name: Kevin
+context: ./context
+
+...
 ```
+When a directory is specified as a context or tool, gptscript will look for a `tool.gpt` in that directory and use it.
+
+The above example context tool has several interesting features:
+
+The `share context` directive is used to include additional context from other tools or files into the tool that referenced this context. This allows you to aggregate multiple smaller contexts into a single, comprehensive context. Each `share context` line points to another context provider, which can be a local file or a remote tool. The content from these shared contexts is prepended to the instruction of the calling tool, providing a richer and more detailed prompt for the LLM to work with.
```

Comment:
We call preamble fields "directives" here and "tool parameters" elsewhere. Is it worth standardizing the verbiage across the docs?

Reply:
Would be nice to standardize...ive also used the word stanza. "directive" here was actually a new one generated by gpt 😅 

I do not like tool parameters because parameter is one of the parameters.

Reply:
im actually going to be adding more commits to this with other changes (with the goal of standardizing some other stuff).

Reply:
> I do not like tool parameters because parameter is one of the parameters

hard concur

> im actually going to be adding more commits to this with other changes (with the goal of standardizing some other stuff)

sweet, thanks

Reply:
Just to throw my vote in the hat, I think "tool field", "tool preamble field", or "tool definition field" are some descriptive options that are different _enough_ from "tool parameters" to prevent confusion.